yb-voyager/src/tgtdb/yugabytedb.go

```diff
@@ -1289,3 +1289,115 @@ func (yb *TargetYugabyteDB) GetImportedSnapshotRowCountForTable(tableName string
 	log.Infof("total row count for snapshot import of table %s: %d", tableName, snapshotRowCount)
 	return snapshotRowCount, nil
 }
+
+func (yb *TargetYugabyteDB) GetGeneratedAlwaysIdentityColumnNamesForTable(table string) ([]string, error) {
+	schema := yb.getTargetSchemaName(table)
+	// TODO: handle case-sensitivity correctly
+	if utils.IsQuotedString(table) {
+		table = table[1 : len(table)-1]
+	} else {
+		table = strings.ToLower(table)
+	}
+	query := fmt.Sprintf(`SELECT column_name FROM information_schema.columns where table_schema='%s' AND
+		table_name='%s' AND is_identity='YES' AND identity_generation='ALWAYS'`, schema, table)
+	var identityColumns []string
+	err := yb.connPool.WithConn(func(conn *pgx.Conn) (bool, error) {
+		rows, err := conn.Query(context.Background(), query)
+		if err != nil {
+			log.Errorf("error in querying identity columns for table %s: %v", table, err)
+			return false, fmt.Errorf("error in querying identity columns for table %s: %w", table, err)
+		}
+		defer rows.Close()
+		for rows.Next() {
+			var colName string
+			err = rows.Scan(&colName)
+			if err != nil {
+				log.Errorf("error in scanning identity column name: %v", err)
```

Comment:
No need of repeating error messages in the log if we're already returning an error with the same message. The error will anyway be logged by the top level function e.g. `main` or `utils.ErrExit()`.

Reply:
Simplified the error message a bit in these cases.
But here adding more context to it by telling whether the error was during the `querying` or `scanning` part, so will keep that part